src/test/cloudWatchLogs/registry/logStreamRegistry.test.ts

```diff
@@ -108,9 +108,9 @@ describe('LogStreamRegistry', async function () {
 
     beforeEach(function () {
         map = new Map<string, CloudWatchLogsData>()
-        map.set(registeredUri.path, stream1)
-        map.set(shorterRegisteredUri.path, simplerStream)
-        map.set(newLineUri.path, newLineStream)
+        map.set(uriToKey(registeredUri), stream1)
+        map.set(uriToKey(shorterRegisteredUri), simplerStream)
+        map.set(uriToKey(newLineUri), newLineStream)
         registry = new LogStreamRegistry(new CloudWatchLogsSettings(config), map)
```

Comment:
Keying by the URI object itself was not working for me. It seemed to cause some issues. I can try to come up with a way to make this more clear that `UriToKey` is expected.

Reply:
> Keying by the URI object itself was not working for me

oops right that won't work, a string key is correct. 

> it's only when adding maps manually that you would "want" it. It doesn't break without it, assuming you give unique strings as keys.

Ok, is the "manual" operation private or exposed? Would be nice to avoid exposing it, even in the constructor as shown here. Inventing a custom class for this is probably too much indirection, but maybe instead of accepting a Map in the ctor, can we have a `LogStreamRegistry.add(vscode.Uri, <stream>)` method (if one doesn't already exist)? Then instead of doing `map.set()` here, it would look like:

```typescript
registry = new LogStreamRegistry(new CloudWatchLogsSettings(config))
registry.add(registeredUri, stream1)
registry.add(shorterRegisteredUri, simplerStream)
registry.add(newLineUri, newLineStream)
```

Reply:
I figured out a way to not expose it. So usually we could use `registry.registerLog` to hide this function, but since we have tests for that in the file, we are trying to wait on officially register the logs. However, if we make the `setLog` method public, this is exactly what you are describing.